plugins/github/komand_github/actions/add_collaborator/action.py

```diff
@@ -1,35 +1,67 @@
-import komand
-from .schema import AddCollaboratorInput, AddCollaboratorOutput
-
-# Custom imports below
 import requests
+import insightconnect_plugin_runtime
+import urllib.parse
+
+from insightconnect_plugin_runtime.exceptions import PluginException
+from komand_github.actions.add_collaborator.schema import (
+    AddCollaboratorInput,
+    AddCollaboratorOutput,
+    Input,
+    Output,
+    Component,
+)
 
 
-class AddCollaborator(komand.Action):
+class AddCollaborator(insightconnect_plugin_runtime.Action):
     def __init__(self):
         super(self.__class__, self).__init__(
             name="add_collaborator",
-            description="Adds a user as a collaborator to a Github repository",
+            description=Component.DESCRIPTION,
             input=AddCollaboratorInput(),
             output=AddCollaboratorOutput(),
         )
 
     def run(self, params={}):
-        r = requests.session()
+        organization = urllib.parse.quote(params.get(Input.ORGANIZATION))
+        repository = urllib.parse.quote(params.get(Input.REPOSITORY))
+        username = urllib.parse.quote(params.get(Input.USERNAME))
+        permission = params.get(Input.PERMISSION)
+
+        url = requests.compat.urljoin(
+            self.connection.api_prefix, f"/repos/{organization}/{repository}/collaborators/{username}"
+        )
+
         try:
-            results = r.put(
-                "https://api.github.com/repos/{}/{}/collaborators/{}".format(
-                    params.get("organization"),
-                    params.get("repository"),
-                    params.get("username"),
-                ),
-                auth=self.connection.basic_auth,
-                params={"permission": params.get("permission")},
+            results = requests.put(
+                url=url, headers=self.connection.auth_header, params={"permission": permission}, timeout=60
             )
-            if results.status_code == 204:
-                return {"results": "User is already a collaborator"}
-            elif results.status_code == 201:
-                return {"results": results.json()}
-        except Exception as e:
-            self.logger.error("An error has occurred while adding collaborator: ", e)
-            raise
+            if results.status_code == 201:
```

Comment:
Similarly is there a way we could gather the status code handlers together or default some of their text? We might be able to use the values stored in the SDK if they are suitable.

Reply:
I have added all generic error handling to the util file